packages/core/src/internal/fetchSource.ts

```diff
@@ -172,7 +172,17 @@ async function* fetchOperation(
     } else if (!/text\//i.test(contentType)) {
       results = parseJSON(response);
     } else {
-      throw new Error(await response.text());
+      const text = await response.text();
+      try {
+        results = JSON.parse(text);
```

Comment:
Small issue here, `results` is expected to be an async iterator (for the for-await loop below). It's likely not warning here since the result of the parse is `any` and it may still appear to work, but that likely will run into a runtime error 🤔